### ISSUE TITLE
[SDK46][expo-constants] build failing on getAppConfig.js

### ISSUE THREAD

**Issue by robrechtme** [expo/expo#18490]
### Summary

After upgrading a bare workflow to SDK 46, I get the following error:

```
Node found at: /Users/robrechtme/.asdf/shims/node
node:internal/fs/utils:344
    throw err;
    ^

Error: ENOENT: no such file or directory, open '/Users/robrechtme/Library/Developer/Xcode/DerivedData/MyApp-azpdixtdlkszolgclavvwztubvel/Build/Intermediates.noindex/ArchiveIntermediates/MyApp/app.config'
    at Object.openSync (node:fs:585:3)
    at Object.writeFileSync (node:fs:2153:35)
    at Object.<anonymous> (/Users/robrechtme/code/my-app/node_modules/expo-constants/scripts/getAppConfig.js:22:4)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47 {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/Users/robrechtme/Library/Developer/Xcode/DerivedData/MyApp-azpdixtdlkszolgclavvwztubvel/Build/Intermediates.noindex/ArchiveIntermediates/MyApp/app.config'
}
Command PhaseScriptExecution failed with a nonzero exit code
```

The error seems to be coming from [`expo-constants/scripts/getAppConfig.js#L22`](https://github.com/expo/expo/blob/main/packages/expo-constants/scripts/getAppConfig.js#L22). When I comment out that line, the build succeeds.

The error only occurs when archiving the app. During development builds it is ok.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

iOS

### SDK Version (managed workflow only)

46

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 12.5
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.13.1 - ~/.asdf/installs/nodejs/16.13.1/bin/node
      Yarn: 1.22.18 - ~/.asdf/shims/yarn
      npm: 8.6.0 - ~/.asdf/installs/nodejs/16.13.1/.npm/bin/npm
      Watchman: 2022.07.04.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /Users/robrechtme/.asdf/shims/pod
    SDKs:
      Android SDK:
        Android NDK: 22.1.7171670
    IDEs:
      Android Studio: 2020.3 AI-203.7717.56.2031.7935034
      Xcode: /undefined - /usr/bin/xcodebuild
    npmPackages:
      expo: 46.0.2 => 46.0.2
      react: 18.0.0 => 18.0.0
      react-native: 0.69.3 => 0.69.3
    npmGlobalPackages:
      eas-cli: 0.44.1
      expo-cli: 6.0.1
    Expo Workflow: bare

### Reproducible demo

No demo sorry

**Comment by robrechtme** [comment]
It seems like the `/Users/robrechtme/Library/Developer/Xcode/DerivedData/MyApp-azpdixtdlkszolgclavvwztubvel/Build/Intermediates.noindex/ArchiveIntermediates/MyApp/` directory does not exist, only `/Users/robrechtme/Library/Developer/Xcode/DerivedData/MyApp-azpdixtdlkszolgclavvwztubvel/Build/Intermediates.noindex/ArchiveIntermediates/MyApp (staging)/`

**Comment by brentvatne** [comment]
@robrechtme - can you try to provide a reproducible example please? we'd be happy to investigate if so

**Comment by MHV1** [comment]
Exact same thing happened here right after upgrading to React Native 0.69.3 and Expo 46.0.2. Environment configuration is pretty much the same as @robrechtme's, except we are using Node 16.14.0 and Xcode 13.4.1, and also on a bare project. Only difference is that we noticed this on the CI build. The build succeeds but the archiving phase doesn’t. Local debug build works just fine.

Here's the log output:

`/bin/sh -c /Users/runner/Library/Developer/Xcode/DerivedData/MyApp-amqqafyylswerqgmkfoyroszsdlz/Build/Intermediates.noindex/ArchiveIntermediates/MyApp\\\ Test/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/EXConstants.build/Script-46EB2E00025FE0.sh

Node found at: /Users/runner/hostedtoolcache/node/16.14.0/x64/bin/node
node:internal/fs/utils:344
    throw err;
    ^

Error: ENOENT: no such file or directory, open '/Users/runner/Library/Developer/Xcode/DerivedData/MyApp-amqqafyylswerqgmkfoyroszsdlz/Build/Intermediates.noindex/ArchiveIntermediates/MyApp/app.config'
    at Object.openSync (node:fs:585:3)
    at Object.writeFileSync (node:fs:2155:35)
    at Object.<anonymous> (/Users/runner/work/1/s/MyApp/node_modules/expo-constants/scripts/getAppConfig.js:22:4)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1155:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47 {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/Users/runner/Library/Developer/Xcode/DerivedData/MyApp-amqqafyylswerqgmkfoyroszsdlz/Build/Intermediates.noindex/ArchiveIntermediates/MyApp/app.config'
}
Command PhaseScriptExecution failed with a nonzero exit code

** ARCHIVE FAILED **


The following build commands failed:
	PhaseScriptExecution [CP-User]\ Generate\ app.config\ for\ prebuilt\ Constants.manifest /Users/runner/Library/Developer/Xcode/DerivedData/MyApp-amqqafyylswerqgmkfoyroszsdlz/Build/Intermediates.noindex/ArchiveIntermediates/MyApp\ Test/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/EXConstants.build/Script-46EB2E00025FE0.sh (in target 'EXConstants' from project 'Pods')
(1 failure)
`

**Comment by robrechtme** [comment]
> Only difference is that we noticed this on the CI build. The build succeeds but the archiving phase doesn’t. Local debug build works just fine.

This was the same for me actually. 

I ended up excluding `expo-constants` since it wasn't used directly in the project:
```jsonc
// package.json
"expo": {
  "autolinking": {
    "exclude": "expo-constants"
  }
},
```

**Comment by rafaelmaeuer** [comment]
I had the same error when archiving an iOS app (bare-workflow) after updating to Expo SDK46. After trying all possible solutions from https://github.com/expo/expo/issues/15809 I did some further investigation. I found out, that in the script path (in intermediate files) there were some strange backslashes:

```
.../ArchiveIntermediates/app-name\\\ \\\(release\\\)/IntermediateBuildFilesPath/...
```

So after renaming the app target from `app-name (release)` to `app-name-release` the issue was gone.
